### DISCUSSION TITLE
Upload a single file with PUT request. Is it supported/recommended?

### DISCUSSION THREAD

**Discussion by Gelembjuk** [fastapi/fastapi#12289] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@f_app.put("/share/{share_name}/file/upload")
def upload_file(request: Request, share_name:str, file: UploadFile = File(...)):
    path = request.headers.get("x-file-path","")
    # set stream to the input. We will read from there in the controller
    request.state.router.set_stream(file.file)
    
    return UploadAPIEndpoint(request.state.api_app).action("put",
                            share_name = share_name, path = path)
```

### Description

In my REST API i want to support upload of files with PUT request.

When i search FastAPI documentation i see only examples with POST request and multipart forms, like this one https://fastapi.tiangolo.com/tutorial/request-files/

I have created the single example for PUT request. Please see my code. It works with a curl test

```
curl "http://10.100.23.30:8080/share/MyShare/file/upload" -X PUT -H 'X-File-Path: F1/xxx.rb' -F 'file=@/Users/x/Temp/test1.rb' -H "Authorization: Bearer **MYTOKEN**"
```

My problem is that this example doesn't work with the Postman tool.
I have posted this problem on stackoverflow but it is marked as duplicate and connected to a POST example.
I tried tor explain that it is not about POST but PUT. However, all comments refer me to POST examples.

Is PUT officially supported in FastAPI for uploads?

Maybe you know the answer why Postman does not work with PUT upload?
There is my original stackoverflow question 
https://stackoverflow.com/questions/79030921/postman-fails-put-request-to-fastapi-to-upload-a-file?noredirect=1

### Operating System

Linux

### FastAPI Version

latest

### Pydantic Version

latest

### Python Version

3.11

**Comment by Kfir-G** [comment]
1. **PUT Request Support**: I couldn't find any official documentation on this.

2. **Postman Configuration**: Here’s what you can try:
   - Set the request method to **PUT**.
   - In the **Body** tab, choose **form-data**.
     - For the **Key**, use `file` (this needs to match your FastAPI function).
     - For the **Value**, select the file you want to upload.
   - Don’t forget to add any necessary headers, like `Authorization` or `X-File-Path`.

3. **Troubleshooting**: If you get a `422 Unprocessable Entity` error, it usually means that the key in Postman isn’t matching the parameter in your FastAPI or there’s an issue with the file type validation.

**Comment by Kfir-G** [comment]
By the way, I recommend removing the bearer token from your question.

**Reply by Gelembjuk** [reply]
Thanks. That token is just from testing. But anyway, good practice is to remove. I will do

**Comment by YuriiMotov** [comment, marked as answer]
There is no difference between PUT and POST in this case.
The problem is most likely in the configuration of your request on Postman
